nemo/utils/exp_manager.py

```diff
@@ -96,7 +96,10 @@ class CallbackParams:
     save_top_k: Optional[int] = 3
     save_weights_only: Optional[bool] = False
     mode: Optional[str] = "min"
+    auto_insert_metric_name: bool = True
```

Comment:
What does this bit do? And how does it interact with specifying a filename that already has metrics specified like this in `megatron_gpt_config.yaml`. 

Ex: `filename: 'megatron_gpt--{val_loss:.2f}-{step}-{consumed_samples}`

Reply:
When this is true, it will not break anything and behaves the same as until now. It's already true by default on PTL. When it's true, PTL will replace {val_loss:.2f} with something like val_loss=0.12-.
The equal (=) sign and the dash (-) signs are what PTL auto-inserts to the metric name. So it won't change anything.

However, when set to False, you can specify the name of the checkpoint manually, to replace some of those characters with any character you want. This can help avoid logging issues with W&B, TB, or other systems that don't like the = and - characters, so it gives you full control over the checkpoint name.